tests/Doctrine/Tests/ORM/Functional/Ticket/GH9109Test.php

```diff
@@ -0,0 +1,201 @@
+
+        $this->_em->persist($user);
+        $this->_em->persist($product);
+        $this->_em->flush();
+
+        $product->addBuyer($user);
+
+        $this->_em->persist($product);
+        $this->_em->flush();
+
+        $this->_em->clear();
+
+        $persistedProduct = $this->_em->find(GH9109Product::class, $product->getId());
+
+        // assert Product was persisted
+        self::assertInstanceOf(GH9109Product::class, $persistedProduct);
+        self::assertEquals($productTitle, $persistedProduct->getTitle());
+
+        // assert Product has a Buyer
+        $criteria = Criteria::create();
+        $count = $persistedProduct->getBuyers()->matching($criteria)->count();
+        self::assertEquals(1, $count);
+
+        // assert NOT QUOTED WORK with findOneBy
+        $user = $userRepository->findOneBy(['lastName' => $userLastName]);
+        self::assertInstanceOf(GH9109User::class, $user);
+        self::assertEquals($userLastName, $user->getLastName());
+
+        // assert NOT QUOTED WORK with Criteria
+        $criteria = Criteria::create();
+        $criteria->where($criteria->expr()->eq('lastName', $userLastName));
+        $matchedUserLastName = $persistedProduct->getBuyers()->matching($criteria)->first();
+        self::assertInstanceOf(GH9109User::class, $matchedUserLastName);
+        self::assertEquals($userLastName, $matchedUserLastName->getLastName());
+
+        // assert QUOTED WORK with findOneBy
+        $user = $userRepository->findOneBy(['firstName' => $userFirstName]);
+        self::assertInstanceOf(GH9109User::class, $user);
+        self::assertEquals($userFirstName, $user->getFirstName());
+
+        // assert QUOTED DOES WORK with Criteria
+        $criteria = Criteria::create();
+        $criteria->where($criteria->expr()->eq('firstName', $userFirstName));
+        $matchedUserFirstName = $persistedProduct->getBuyers()->matching($criteria)->first();
+        self::assertInstanceOf(GH9109User::class, $matchedUserFirstName);
+        self::assertEquals($userFirstName, $matchedUserFirstName->getFirstName());
+    }
+}
+
+/**
+ * @Entity
+ */
+class GH9109Product
+{
+    /**
+     * @var int $id
+     * @Column(name="`id`", type="integer")
+     * @Id
+     * @GeneratedValue(strategy="IDENTITY")
+     */
+    private $id;
+
+    /**
+     * @var string $title
+     * @Column(name="`title`", type="string", length=255)
+     */
+    private $title;
+
+    /**
+     * @var Collection|GH9109User[]
+     * @psalm-var Collection<int, GH9109User>
+     * @ManyToMany(targetEntity="GH9109User")
+     */
+    private $buyers;
+
+    public function __construct()
+    {
+        $this->buyers = new ArrayCollection();
+    }
+
+    public function getId(): int
+    {
+        return $this->id;
+    }
+
+    public function setTitle(string $title): void
+    {
+        $this->title = $title;
+    }
+
+    public function getTitle(): string
+    {
+        return $this->title;
+    }
+
+    /**
+     * @return Collection
```

Comment:
Please use annotations similar to those line 112 - 113

Reply:
done